ipatests/test_integration/test_sssd.py

```diff
@@ -352,3 +355,31 @@ def test_external_group_member_mismatch(self, user_origin):
             assert user_origin == 'ipa'
         finally:
             master.run_command(['ipa', 'group-del', 'ext-ipatest'])
+
+    def test_trustdomain_disable(self):
+        """Test that users from disabled trustdomains can not use ipa resources
+
+        This is a regression test for sssd bug:
+        https://pagure.io/SSSD/sssd/issue/4078
+        """
+        ad_domain_name = self.ad.domain.name
+        ad_subdomain_name = self.child_ad.domain.name
+        test_user = 'subdomaintestuser@{0}'.format(ad_subdomain_name)
+
+        try:
+            # verify the user can be retreived initially
+            self.master.run_command(['id', test_user])
+
+            self.master.run_command(['ipa', 'trustdomain-disable',
+                                     ad_domain_name, ad_subdomain_name])
+            tasks.clear_sssd_cache(self.master)
+            res = self.master.run_command(['id', test_user], raiseonerr=False)
```

Comment:
I would also add another check to this test, to make sure that users from the parent domain are still able to access IdM resources when the child domain is disabled.

Reply:
Added two checks:
1. that users from parent and ipa domain can be retrieved when trust subdomain is disabled
2. that subdomain user can be retrieved after re-enabling trustdomain